intermediate_source/inductor_debug_cpu.py

```diff
@@ -0,0 +1,617 @@
+    #pragma omp parallel num_threads(32)
+    {
+        {
+            #pragma omp for 
+            for(long i0=static_cast<long>(0L); i0<static_cast<long>(16384L); i0+=static_cast<long>(1L))
+            {
+                for(long i1=static_cast<long>(0L); i1<static_cast<long>(512L); i1+=static_cast<long>(8L))
+                {
+                    auto tmp0 = at::vec::Vectorized<float>::loadu(in_ptr0 + static_cast<long>(i1 + (512L*i0)));
+                    auto tmp1 = at::vec::Vectorized<float>::loadu(in_ptr1 + static_cast<long>(i1));
+                    auto tmp3 = at::vec::Vectorized<float>::loadu(in_out_ptr0 + static_cast<long>(i1 + (512L*i0)));
+                    auto tmp5 = at::vec::Vectorized<float>::loadu(in_ptr2 + static_cast<long>(i1));
+                    auto tmp7 = at::vec::Vectorized<float>::loadu(in_ptr3 + static_cast<long>(i1));
+                    auto tmp2 = tmp0 + tmp1;
+                    auto tmp4 = tmp2 + tmp3;
+                    auto tmp6 = tmp4 * tmp5;
+                    auto tmp8 = tmp6 + tmp7;
+                    tmp8.store(in_out_ptr0 + static_cast<long>(i1 + (512L*i0)));
+                }
+            }
+        }
+    }
+}''')
+
+######################################################################
+# From the generated code above, we can see this kernel has done a typical `Loop Fusion <https://en.wikipedia.org/wiki/Loop_fission_and_fusion>`_ on ``[add, add, mul, add]``.
+# This is a memory-bound bottle neck preventing good performance. To get a more intuitive feeling about this optimization, 
+# we can infer the sizes and stride of the inputs and further benchmark this ``[add, add, mul, add]`` pattern.
+
+def func(arg_0, arg_1, arg_2, arg_3, arg_4):
+    add_0 = arg_0 + arg_1
+    add_1 = add_0 + arg_2
+    mul_1 = add_1 * arg_3
+    add_2 = mul_1 + arg_4
+    arg_2 = add_2
+    return arg_2
+
+arg_0 = torch.rand(16384, 512)
+arg_1 = torch.rand(1, 512)
+arg_2 = torch.zeros(16384, 512)
+arg_3 = torch.rand(1, 512)
+arg_4 = torch.rand(1, 512)
+
+input = (arg_0, arg_1, arg_2, arg_3, arg_4)
+inductor_func = torch.compile(func)
+with torch.no_grad():
+    inductor_func(*input)
+
+import timeit
+NUM_ITERS=1000
+with torch.no_grad():
+    # warmup
+    for _ in range(10):
+        func(*input)
+    eager_t = timeit.timeit("func(*input)", number=NUM_ITERS, globals=globals())
+
+with torch.no_grad():
+    # warmup
+    for _ in range(10):
+        inductor_func(*input)
+    inductor_t = timeit.timeit("inductor_func(*input)", number=NUM_ITERS, globals=globals())
+print(f"eager use: {eager_t * 1000 / NUM_ITERS} ms/iter")
+print(f"inductor use: {inductor_t * 1000 / NUM_ITERS} ms/iter")
+print(f"speed up ratio: {eager_t / inductor_t}")
+
+######################################################################
+# Output:
+#
+# .. code-block:: shell
+#
+#     eager use: 5.780875144992024 ms/iter
+#     inductor use: 0.9588955780491233 ms/iter
+#     speed up ratio: 6.0286805751604735
+#
```

Comment:
This will be generated in the final [HTML]( I see duplicate outputs in the resulting HTML: https://docs-preview.pytorch.org/pytorch/tutorials/2430/intermediate/inductor_debug_cpu.html#performance-profiling ). Can we remove this block or state in which env you get this numbers if it's important to show specifically these results?

Reply:
Hi, @svekars. Will this automatically generate results running from time to time and get different outputs? Or will it be frozen after PR is merged?